js/MineSweeper.js

```diff
@@ -397,6 +409,18 @@ jQuery(function ($) {
                 var dim_y = parseInt($('#dim_y').val(), 10);
                 var num_mines = parseInt($('#num_mines').val(), 10);
 
+                // rationalise options JIC
+                if (isNaN(dim_x) || (dim_x == 0)) dim_x = 1;
```

Comment:
these if/else blocks need to be like:
```javascript
if (isNaN(dim_x) || (dim_x == 0)) {
    dim_x = 1;
}
```

note the brackets are required, and expressions should be on their own line for readability.